db/system_keyspace.cc

```diff
@@ -233,12 +233,15 @@ schema_ptr system_keyspace::topology() {
             .with_column("request_id", timeuuid_type)
             .with_column("ignore_nodes", set_type_impl::get_instance(uuid_type, true), column_kind::static_column)
             .with_column("new_cdc_generation_data_uuid", timeuuid_type, column_kind::static_column)
+            .with_column("new_keyspace_rf_change_ks_name", utf8_type, column_kind::static_column)
+            .with_column("new_keyspace_rf_change_rf_per_dc", map_type_impl::get_instance(utf8_type, utf8_type, false), column_kind::static_column)
```

Comment:
I believe that new columns should be added last, for backward compatibility reasons.

Reply:
Also, using (insert/update) the new columns should be only based on a cluster feature, to support upgrade rollback.

Reply:
it means that adding a column should be conditional?
I don't see anything like that in system_keyspace.cc, but should be fairly easy to implement

Reply:
The additional columns can be returned here unconditionally. Just writing data to them should depend on a cluster feature.

See https://github.com/scylladb/scylladb/pull/18458 for example.

Reply:
we discussed that again and since the use case is altering a keyspace that uses tablets, tablets had to be enabled to begin with, so that settles the upgrade-related concern.  So I retract this comment.